examples/apps/cli/main.c

```diff
@@ -77,8 +77,10 @@ static const otCliCommand kCommands[] = {
     /*
      * The CLI command `nodeidfilter` only works for simulation in real time.
      * The usage of the command `nodeidfilter`:
-     *     - `nodeidfilter deny <nodeid>`:  It denies the connection to a specified node.
-     *     - `nodeidfilter clear`:          It restores the filter state to default.
+     *     - `nodeidfilter deny <nodeid>` :  It denies the connection to a specified node (use as deny-list).
+     *     - `nodeidfilter allow <nodeid> :  It allows the connection to a specified node (use as allow-list),
```

Comment:
It seems we do not support configuring allow/deny for each individual nodeid. Should we separate the allow/deny config from the list config?

Reply:
It behaves like `macfilter`, i.e. it can be used either as an allow-list or a deny-list . Once the filter is cleared, the first call to `nodeidfilter allow` or `nodeidfilter deny` determines its mode. Then we can only use same sub-command, e.g., if we first call `nodeidfilter allow`, then a subsequent `nodeidfilter deny` will give an `InvalidState` error .

I updated the comment here in the code to describe this in more detail.

Reply:
`macfilter` has separate commands for configuring allow/deny vs. managing the list.

But I think current model can work as well.